archinstall/lib/disk/device_handler.py

```diff
@@ -341,16 +341,34 @@ def _perform_partitioning(
 			# the partition has a real path now as it was created
 			part_mod.dev_path = Path(partition.path)
 
-			info = get_lsblk_info(part_mod.dev_path)
-
-			if not info.partuuid:
-				raise DiskError(f'Unable to determine new partition uuid: {part_mod.dev_path}')
+			info = self._fetch_partuuid(part_mod.dev_path)
 
 			part_mod.partuuid = info.partuuid
 			part_mod.uuid = info.uuid
 		except PartitionException as ex:
 			raise DiskError(f'Unable to add partition, most likely due to overlapping sectors: {ex}') from ex
 
+	def _fetch_partuuid(self, path: Path) -> LsblkInfo:
+		attempts = 3
+		info: Optional[LsblkInfo] = None
+
+		for attempt_nr in range(attempts):
+			self.partprobe(path)
```

Comment:
Can we move this `partprobe()` out right before the loop?
AFAIK, calling `partprobe()` once will initiate the kernel to look for new ID's, and it will take X amount of time and the kernel will not stop partprob-ing until all ID's are populated. Calling `partprobe()` again will just re-start that process.

Reply:
Right, so we want to kick off partprobe then wait for x seconds and the run lsblk?

Reply:
That's the curse of `partprobe()`, it doesn't block.
Which means you can't rely on "when partprobe is done, all the data will be there".

At least that's my experience with it.

Reply:
In that case I'll update the PR